File: ureka-framework-main/ureka_framework/logic/stage_worker/msg_generator.py
# Data Model (RAM)
from ureka_framework.model.shared_data import SharedData

# Data Model (Message)
from ureka_framework.model.message_model.u_ticket import u_ticket_to_jsonstr
from ureka_framework.model.message_model.r_ticket import r_ticket_to_jsonstr

# Resource (Logger)
from ureka_framework.resource.logger.simple_logger import simple_log

# Resource (Measurer)
from ureka_framework.resource.logger.simple_measurer import measure_worker_func

# Stage Worker
from ureka_framework.logic.stage_worker.msg_generator_u_ticket import UTicketGenerator
from ureka_framework.logic.stage_worker.msg_generator_r_ticket import RTicketGenerator

# Measure Helper
from ureka_framework.logic.stage_worker.measure_helper import MeasureHelper


class MsgGenerator:
    def __init__(
        self,
        shared_data: SharedData,
        measure_helper: MeasureHelper,
    ) -> None:
        self.shared_data = shared_data
        self.measure_helper = measure_helper

    ######################################################
    # [STAGE: (G)] Generate Message
    ######################################################
    @measure_worker_func
    def _generate_xxx_u_ticket(self, arbitrary_dict: dict) -> str:
        simple_log(
            "info",
            f"+ {self.shared_data.this_device.device_name} is generating u_ticket...",
        )

        u_ticket_generator = UTicketGenerator(
            self.shared_data.this_device,
            self.shared_data.this_person,
            self.shared_data.device_table,
        )
        generated_u_ticket = u_ticket_generator.generate_arbitrary_u_ticket(
            arbitrary_dict
        )
        generated_u_ticket_json = u_ticket_to_jsonstr(generated_u_ticket)

        return generated_u_ticket_json

    @measure_worker_func
    def _generate_xxx_r_ticket(self, arbitrary_dict: dict) -> str:
        simple_log(
            "info",
            f"+ {self.shared_data.this_device.device_name} is generating r_ticket...",
        )

        r_ticket_generator = RTicketGenerator(
            self.shared_data.this_device,
            self.shared_data.this_person,
            self.shared_data.device_table,
        )
        generated_r_ticket = r_ticket_generator.generate_arbitrary_r_ticket(
            arbitrary_dict
        )
        generated_r_ticket_json = r_ticket_to_jsonstr(generated_r_ticket)

        return generated_r_ticket_json

File: ureka-framework-java/app/src/main/java/ureka/framework/logic/stage_worker/MsgGenerator.java
package ureka.framework.logic.stage_worker;

import java.util.Map;

import ureka.framework.model.SharedData;
import ureka.framework.model.message_model.RTicket;
import ureka.framework.model.message_model.UTicket;
import ureka.framework.resource.logger.SimpleLogger;
import ureka.framework.resource.logger.SimpleMeasurer;

public class MsgGenerator {
    private SharedData sharedData;
    private MeasureHelper measureHelper;

    public MsgGenerator(SharedData sharedData, MeasureHelper measureHelper) {
        this.sharedData = sharedData;
        this.measureHelper = measureHelper;
    }

    // [STAGE: (G)] Generate Message
    public String generateXxxUTicket(Map<String, String> arbitraryDict) {
        return SimpleMeasurer.measureWorkerFunc(this::_generateXxxUTicket, arbitraryDict);
    }

    private String _generateXxxUTicket(Map<String, String> arbitraryDict) {
        SimpleLogger.simpleLog("info",this.sharedData.getThisDevice().getDeviceName() + " is generating u_ticket...");

        MsgGeneratorUTicket uTicketGenerator = new MsgGeneratorUTicket(
                this.sharedData.getThisDevice(),
                this.sharedData.getThisPerson(),
                this.sharedData.getDeviceTable()
        );
        UTicket generatedUTicket = uTicketGenerator.generateArbitraryUTicket(arbitraryDict);
        String generatedUTicketJson = UTicket.uTicketToJsonStr(generatedUTicket);

        return generatedUTicketJson;
    }

    public String generateXxxRTicket(Map<String, String> arbitraryDict) {
        return SimpleMeasurer.measureWorkerFunc(this::_generateXxxRTicket, arbitraryDict);
    }

    private String _generateXxxRTicket(Map<String, String> arbitraryDict) {
        SimpleLogger.simpleLog("info",this.sharedData.getThisDevice().getDeviceName() + " is generating r_ticket...");

        MsgGeneratorRTicket rTicketGenerator = new MsgGeneratorRTicket(
                this.sharedData.getThisDevice(),
                this.sharedData.getThisPerson(),
                this.sharedData.getDeviceTable()
        );
        RTicket generatedRTicket = rTicketGenerator.generateArbitraryRTicket(arbitraryDict);
        String generatedRTicketJson = RTicket.rTicketToJsonStr(generatedRTicket);

        return generatedRTicketJson;
    }

}
